packages/edit-site/src/components/editor/use-editor-title.js

```diff
@@ -4,33 +4,49 @@
 import { _x, sprintf } from '@wordpress/i18n';
 import { useSelect } from '@wordpress/data';
 import { store as coreStore } from '@wordpress/core-data';
-import { store as editorStore } from '@wordpress/editor';
 import { decodeEntities } from '@wordpress/html-entities';
+import { privateApis as editorPrivateApis } from '@wordpress/editor';
 
 /**
  * Internal dependencies
  */
 import useTitle from '../routes/use-title';
 import { POST_TYPE_LABELS, TEMPLATE_POST_TYPE } from '../../utils/constants';
+import { unlock } from '../../lock-unlock';
+
+const { getTemplateInfo } = unlock( editorPrivateApis );
 
 function useEditorTitle( postType, postId ) {
 	const { title, isLoaded } = useSelect(
 		( select ) => {
 			const { getEditedEntityRecord, hasFinishedResolution } =
 				select( coreStore );
-			const { __experimentalGetTemplateInfo: getTemplateInfo } =
-				select( editorStore );
+
 			const _record = getEditedEntityRecord(
 				'postType',
 				postType,
 				postId
 			);
+
+			const templateAreas =
+				select( coreStore ).getEntityRecord( 'root', '__unstableBase' )
+					?.default_template_part_areas || [];
+
+			const templateTypes =
+				select( coreStore ).getEntityRecord( 'root', '__unstableBase' )
+					?.default_template_types || [];
```

Comment:
This can be combined into a single call. The `coreStore` selectors are also extracted above; let's add `getEntityRecord` to that list. 

Bit ugly bug efficient 😄

```js
const {
	default_template_part_areas: templateAreas = [],
	default_template_types: templateTypes = [],
} = getEntityRecord( 'root', '__unstableBase' ) ?? {};
```

Reply:
Fixed with e395bf67bd06b772e8057f1c78c21b78bbf2630d